conformance/tests/httproute-response-header-modifier.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, HTTPRouteResponseHeaderModifier)
+}
+
+var HTTPRouteResponseHeaderModifier = suite.ConformanceTest{
```

Comment:
It's really great to add a conformance test up front like this, but I think we need to hide it behind a "SupportedFeature" so it's opt-in while it's still in the experimental channel: https://github.com/kubernetes-sigs/gateway-api/blob/48cf5d9e4756dd65be21fa26e17e3627903c070a/conformance/utils/suite/suite.go#L42-L55

Reply:
Thanks for bringing this up, I was going to ask about the same thing. I was thinking about what you suggested, but I could not find a way to configure which `SupportedFeature`s should be used while running the conformance tests, as there's no such flag in `conformance/utils/flags/flags.go`.
Atm, it seems to be hardcoded here:
https://github.com/kubernetes-sigs/gateway-api/blob/48cf5d9e4756dd65be21fa26e17e3627903c070a/conformance/conformance_test.go#L53-L55

Reply:
Yeah we really should have a flag for this. I think that got missed because so many implementations are running these tests by importing them and running them with the rest of their test suite.

Reply:
Okay, let me create an issue for this then.